### PR TITLE
Math Magicians: tests

### PR DESCRIPTION
**Pull request by brenoxav**
## Math Magicians: tests

- Write unit tests with React Testing Library
- Write unit tests with snapshots with Jest
- Write unit tests for the files operate.js and calculate.js using Jest
- Create unit tests for all React components
- Use Jest snapshots to test the components
- Use React Testing Library to simulate user interaction
- All tests should pass

### CODE REVIEW

File: src/logic/__tests__/calculate.test.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
+import calculate from '../calculate';
+
+describe('calculate', () => {
+  test('it sould return a null object when AC button is used', () => {
```
Comment (bellom) [inline src/logic/__tests__/calculate.test.js]: ### [Optional]
kindly correct the typo `should` not `sould`

File: src/logic/__tests__/calculate.test.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    expect(calculate({ total: '6.5', next: null }, '.')).toEqual({});
+  });
+
+  test("when press '.' it adds the '.' if total doesn't contains '.'", () => {
+    expect(calculate({ total: '6', next: null }, '.')).toEqual({ total: '6.' });
+  });
+
+  test("when press '.' it adds the '0.' if total doesn't exists", () => {
+    expect(calculate({ total: null, next: null }, '.')).toEqual({ total: '0.' });
+  });
```
Comment (bellom) [inline src/logic/__tests__/calculate.test.js:59]: We need to write a unit test for all the functionalities & scenarios in `calculate.js` module.

kindly add unit test for the following case `buttonName === '+/-'`

**Review by bellom — CHANGES_REQUESTED** [review]
## Hi @brenoxav @pountzas  👏 , 

### Awesome job so far!! :+1: 

#### You have done wonderfully well writing unit test and jest snapshot for all the component setting up linter, and also good commit messages 🔥, However, you passed `almost` all the checklist for this project :+1: just a few issues left to work on............ :slightly_smiling_face: 

# Status: Required Changes! ♻️ 

_Kindly check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


### ♻️ Kindly re-submit for review after making all those changes :slightly_smiling_face: Thumps -up await you.....

### Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/logic/__tests__/calculate.test.js (hunk 1 above)
Comment (henatan99) [reply src/logic/__tests__/calculate.test.js]: - This issue is resolved. Well done! :+1:

File: src/logic/__tests__/calculate.test.js (hunk 2 above)
Comment (henatan99) [reply src/logic/__tests__/calculate.test.js:59]: - This issue is well addressed! :clap: :clap:

**Review by henatan99 — APPROVED** [review]
Hi @pountzas && @brenoxav , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**